iOS/Services/StitchLocalMongoDBService/StitchLocalMongoDBService/LocalMongoClient.swift

```diff
@@ -67,96 +62,125 @@ private final class MongoDBServiceClientProvider: ThrowingServiceClientProvider
             name: .UIApplicationDidReceiveMemoryWarning,
             object: nil
         )
+        // observe when memory warnings are received
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(applicationDidEnterBackground(_:)),
+            name: .UIApplicationDidEnterBackground,
+            object: nil
+        )
+        // observe when memory warnings are received
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(applicationWillTerminate(_:)),
+            name: .UIApplicationWillTerminate,
+            object: nil
+        )
```

Comment:
@edaniels This is where we are hooking into the termination and background sequences to ensure a clean closure of the capi.

Reply:
Let's drop the background task